rust/store/src/lib.rs

```diff
@@ -130,14 +113,15 @@ impl ToTypedValue for i64 {
 
 impl ToTypedValue for f64 {
     fn to_typed_value(&self) -> TypedValue {
-        TypedValue::Double(OrderedFloat(*self))
+        (*self).into()
     }
 }
 
 impl ToTypedValue for Timespec {
     fn to_typed_value(&self) -> TypedValue {
+        // TODO: shouldn't that be / 1000?!
```

Comment:
@fluffyemily I didn't change the logic here, but superficially this looks like a bug…

Reply:
Yes, this is a bug. And quite plainly crashes the code. How has this not been found before? We store `Timespec`'s all the time!

Reply:
Presumably you only ever test this in native code, and you build `libtoodle.a` in release mode, which I imagine permits overflow.